pkg/api/addons.go

```diff
@@ -926,6 +926,18 @@ func (cs *ContainerService) setAddonsConfig(isUpgrade bool) {
 		}
 	}
 
+	// Ensure cloud-node-manager is enabled on appropriate upgrades for Azure Stack cloud
+	if isUpgrade &&
```

Comment:
do we need to check for `to.Bool(cs.Properties.OrchestratorProfile.KubernetesConfig.UseCloudControllerManager) && common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.21.0"`?

The UseCCM flag is overridden already and supported version list starts on 1.21

Reply:
> The UseCCM flag is overridden already

I am assuming that `upgrade.go` is executed first

Reply:
Yes, upgrade.go executed first, I will remove the check for k8s version